backend/notifications.js

```diff
@@ -0,0 +1,32 @@
+const { PrismaClient } = require('@prisma/client');
+const prisma = new PrismaClient();
+
+const checkNotificationConditions = async (patientId) => {
+  const patient = await prisma.patient.findUnique({
+    where: { id: patientId },
+    include: { appointments: true },
+  });
+
+  if (!patient) {
+    throw new Error('Patient not found');
+  }
+
+  const today = new Date();
+  const age = today.getFullYear() - patient.date_of_birth.getFullYear();
```

Comment:
We may want to consider the edge cases where a person is born on a month that hasn't passed on the current year. If they're born Dec 2000, for example, and the current date is Feb 2024, this subtraction would calculate them as 4 since 2024-2000 = 4 even though the person should only be 3 years old.